scripts/user/update-judge.ts

```diff
@@ -0,0 +1,301 @@
+import * as client from '../../web-api/src/persistence/dynamodbClientService';
```

Comment:
Frankly, this script stinks. It is long, and it uses some direct calls to Dynamo instead of persistence abstractions.

That said, our current ad-hoc process of updating a judge is worse, and this script is meant to be a temporary bandaid that has at least been tested without pressing time constraints.

I have preferred to keep the script all in one place rather than breaking it apart because I think locality here will make it easier to improve later (ideally being merged into the DAWSON application logic itself).

In the cases where I used direct calls to Dynamo, no existing persistence methods seemed quite right. They either did too much (e.g., tried to update a Cognito password) or too little (e.g., did not update the section record) and/or were unclear and therefore dangerous to use. (For instance, web-api/src/persistence/dynamo/users/updateUserRecords.ts seems to expect a user with a bar number, but there is nothing else about this function that would suggest a particular user type.) Rather than tie this to persistence methods that were never designed for updating a judge, and rather than create new persistence methods that might be used for unintended purposes in the application, I have preferred direct calls to Dynamo as necessary to keep this script self-contained.

If there are easy things to improve, I am happy to do so; but I think trying to make this rock-solid--especially if our plan is to move to Postgres--is probably not worth the effort.

Reply:
Given that a judge update can potentially impact multiple users (e.g. due to implicit section name change), I think it would be a good idea to have a "dry run" mode that logs out the list of pending actions without executing them.  Or maybe an "are you sure?" prompt with a summary of the pending changes. Feel free to disregard this suggestion if the lift is too much for this temporary script.

Reply:
Let me think about this. I 100% agree that it is a good idea, but I also don't want to commit too much time for something that (I suspect) happens very rarely, especially since I think the worst-case scenario is that one "undoes" the action by using the script again.